lua/telescope/previewers/init.lua

```diff
@@ -1,36 +1,280 @@
+---     `require('telescope.state').get_global_key("last_preview_bufnr")`
+---   - `get_buffer_by_name = function(self, entry)`
+---     Allows you too set a unique name for each buffer. This is used for
+---     caching purpose. `self.state.bufname` will be nil if the entry was
+---     never loaded or the unique name when it was already loaded. So to
+---     make the cache work only load content if
+---     `self.state.bufname ~= entry.your_unique_key`
+---
+--- `self.state` table:
+---   - `self.state.bufnr`
+---     Is the current buffer number, in which you have to write the loaded
+---     content.
+---     Don't create a buffer yourself, otherwise it's not managed by the
+---     buffer_previewer interface and you will probably be better off
+---     writing your own interface.
+---   - self.state.winid
+---     Current window id. Useful if you want to set the cursor to a provided
+---     line number.
+---   - self.state.bufname
+---     Will return the current buffer name, if `get_buffer_by_name` is
+---     defined. nil will be returned if the entry was never loaded or when
+---     `get_buffer_by_name` is not set.
+---
+--- Tips:
+---   - If you want to display content of a terminal job use:
+---     `require('telescope.previewers.utils').job_maker(cmd, bufnr, opts)`
+---       - `cmd` table: for example { 'git', 'diff', entry.value }
+---       - `bufnr` number: in which the content will be written
+---       - `opts` table: with following keys
+---         - `bufname` string: used for cache
+---         - `value` string: used for cache
+---         - `mode` string: either "insert" or "append". "insert" is default
+---         - `env` table: define environment variables. Example:
+---           - `{ ['PAGER'] = '', ['MANWIDTH'] = 50 }`
+---         - `cwd` string: define current working directory for job
+---         - `callback` function(bufnr, content): will be called when job
+---           is done. Content will be nil if job is already loaded.
+---           So you can do highlighting only the first time the previewer
+---           is created for that entry.
+---           Use the returned `bufnr` and not `self.state.bufnr` in callback,
+---           because state can already be changed at this point in time.
+---   - If you want to attach a highlighter use:
+---     - `require('telescope.previewers.utils').highlighter(bufnr, ft)`
+---       - This will prioritize tree sitter highlighting if available for
+---         environment and language.
+---     - `require('telescope.previewers.utils').regex_highlighter(bufnr, ft)`
+---     - `require('telescope.previewers.utils').ts_highlighter(bufnr, ft)`
+---   - If you want to use `vim.fn.search` or similar you need to run it in
+---     that specific buffer context. Do
+--- <pre>
+---       vim.api.nvim_buf_call(bufnr, function()
+---         -- for example `search` and `matchadd`
+---       end)
+---     to achieve that.
+--- </pre>
```

Comment:
```suggestion
---     that specific buffer context. Do
--- <pre>
---       vim.api.nvim_buf_call(bufnr, function()
---         -- for example `search` and `matchadd`
---       end)
--- </pre>
---     to achieve that.
```

Reply:
This is actually fine. `<pre>` isn't supported in a itemize yet so indentation is broken without it :)
Hey but thanks for all the comments :D i have not expected someone to do this i haven't seen before here :)

I will do all changes locally and add you as co-author :)